### PR TITLE
fix recursive propTypes

### PR DESCRIPTION
**Pull request by mathieudutour**
## What

Recursive propTypes needs to be wrapped in a function

cc: @dabbott

### CODE REVIEW

**Comment by ljharb** [comment]
PropTypes can definitely be recursive - you’d need to make the child one a function that wraps the outer one, though, I’ve done it many times.

**Comment by mathieudutour** [comment]
Ah indeed, I didn't know. Thanks

File: compiler/generate-docs/src/components/Layout/Header.js [hunk 1]
```diff
@@ -90,7 +90,7 @@ const FilesPropTypes = PropTypes.objectOf(
     path: PropTypes.string.isRequired,
     title: PropTypes.string,
     order: PropTypes.number,
-    children: FilesPropTypes, // eslint-disable-line no-use-before-define
+    children: () => FilesPropTypes,
```
Comment (ljharb) [inline compiler/generate-docs/src/components/Layout/Header.js]:
```suggestion
    children: (...args) => FilesPropTypes(...args),
```

File: compiler/generate-docs/src/components/Layout/Sidebar.js [hunk 2]
```diff
@@ -167,7 +167,7 @@ const FilesPropTypes = PropTypes.objectOf(
     path: PropTypes.string.isRequired,
     title: PropTypes.string,
     order: PropTypes.number,
-    children: FilesPropTypes, // eslint-disable-line no-use-before-define
+    children: () => FilesPropTypes,
```
Comment (ljharb) [inline compiler/generate-docs/src/components/Layout/Sidebar.js]:
```suggestion
    children: (...args) => FilesPropTypes(...args),
```

**Pull request merged**